### PR TITLE
Add wildcard support for whitelisted email domains for org. signups

### LINKED ISSUE #12846: Remove 10-domain limit from enterprise whitelisting and make it unlimited
*Please explain here below what you were doing when the issue happened*
Many organizations need more than ten subdomains added to the whitelist field in enterprise organizations. Since the domain doesn't cover subdomains, you must add them all manually. This becomes an issue when trying to add more than ten, which is an issue with large orgs and universities. They frequently have hundreds of subdomains.

*Please break down here below all the needed steps to reproduce the issue*
1. add ten domains or subdomains to the whitelisting field in an enterprise organization's Auth tab
2. try to add an eleventh.

*Please describe here below the current result you got*
cannot add an eleventh

*Please describe here below what should be the expected behaviour*
domain would be accepted and you could add more 

*What internet browser (Chrome, Firefox, etc) and version was you using and version*
all

### PR DESCRIPTION
**Pull request by ethervoid**
Related #12846 

This PR adds support for wildcard character (*) in the whitelisted email domains used for organization signups.

### CODE REVIEW

**Comment by Cartofante** [comment]
Can one of the admins verify this patch?

File: app/views/admin/organizations/auth.html.erb [hunk 1]
```diff
@@ -13,10 +13,10 @@
           <div class="FormAccount-rowData">
             <div class="FormAccount-tags js-domains">
               <ul class="CDB-Text CDB-Size-medium FormAccount-tagsList js-domainsList">
-                <span class= "CDB-Text CDB-Size-small u-altTextColor FormAccount-tagsList--placeholder js-placeholder">Only valid domains will be added (ex. carto.com)</span>
+                <span class= "CDB-Text CDB-Size-small u-altTextColor FormAccount-tagsList--placeholder js-placeholder">Only valid domains, wildcard (*) accepted (ex. *.carto.com)</span>
```
Comment (ethervoid) [inline app/views/admin/organizations/auth.html.erb]: @CartoDB/copywriters | @CartoDB/design  are you ok with this change? :)

**Comment by vuxita** [comment]
hey @ethervoid, what's needed from us here? If it's content related this should be for @devonghopkins

**Comment by ethervoid** [comment]
Sorry to ping you but I want to be sure about this change. I've changed the placeholder the for `allowed email domain` component for the organizations so I want a check of that change :)

**Pull request merged**